### DISCUSSION TITLE
Upgrade from 4.x to 5 (9/24/2024) - Compressed Images

### DISCUSSION THREAD

**Discussion by jlpoolen** [LycheeOrg/Lychee#2555] [Q&A] [answered]
I tried to upgrade Lychee last year and found it was going to take some time to change from 4.x to 5, so I let "sleeping dogs lay."  Well, of course, Gentoo's upgrade included a newer version of PHP which revealed errors in the code supporting 4.x.  So I tried to upgrade and staged in a new pristine directory the current high watermark of 5.x  I then pointed to my SQLite database and I ran the script "artisan " plus some parameters recommended in some page or Issue I found relating to upgrades.  I copied my "public" folder from the old 4.x to the new 5.  I confirmed that the SQLite database was updated (I'm very comfortable around databases). 

Now my site seems normal when showing galleries, but when I go to a specific gallery, the images are limited to say 40 pixel high.  Something in the formatting seems to have been altered/left-behind.  

Here is a pastebin of my current 5.x "git status":

https://pastebin.com/CqNJgAdx

What would someone recommend I do to bring my prior collection into alignment?

**Comment by ildyria** [comment]
> but when I go to a specific gallery, the images are limited to say 40 pixel high. Something in the formatting seems to have been altered/left-behind.

Your git status is not going to be very helpful. However we do have a diagnostic page, that would be way more helpful to us so we can help you. Please paste this one.

I would say also to try the `LIVEWIRE_ENABLED=false` in your `.env` as the full front-end v5 is going to be removed soon with v6.

**Reply by jlpoolen** [reply, marked as answer]
@ildyria Thank you.  I found LIVEWIRE_ENABLED=true in my .env file and changed the value to **false**, reconnected to a URL for an album rendered when LIVEWIRE_ENABLED=true was in effect and got an error message such as HTTP not found.  So I went to the top of my site and then the albums displayed differently (the albums displayed in whole previously, but in a different arrangement) and when I clicked through to a selected album, I then was able to view the full content. 
I thought I would try your suggestion first before undertaking a search for the diagnostic page.  If you would like me to revert, run the diagnostic page and submit here for posterity, please let me know.  Otherwise this matter may be considered resolved and closed.  Thank you, again.

**Reply by ildyria** [reply]
No worries.
Hope you will like the version 6. If you feel like taking for a spin, check out the pre-releases.

**Reply by jlpoolen** [reply]
> No worries. Hope you will like the version 6. If you feel like taking for a spin, check out the pre-releases.

If there is a How-to recipe that I can just follow without thinking/worrying, I'll try it.  I know Linux very well and databases, so if it's just a matter of copying my current SQLite database and isolating the version 6, I can do.

**Reply by ildyria** [reply]
It really depends how you are using Lychee: from release file or from git.
But in the end, it is pretty much always the same process if you are using the archive release:
- backup your `uploads` folder.
- backup your database file.

- delete & replace the app folder (it is important to delete because the update may delete some files, having those files left over can screw up the backend while loading: it may look for non-existing dependencies)
- delete & replace the vendor folder
- put back your uploads folder
- put back your database folder
apply `php artisan migrate` and you should be up and running.

The nice things is that you can just make a copy of your sqlite database file and revert easily the DB changes.

We describe it here: https://lycheeorg.github.io/docs/update.html
Though I am not sure the step "Go into the Diagnostic page and click on the Apply migration button." still exists.
